Source/WebCore/rendering/RenderFrameSet.cpp

```diff
@@ -188,26 +189,26 @@ void RenderFrameSet::layOutAxis(GridAxis& axis, std::span<const Length> grid, in
     for (int i = 0; i < gridLen; ++i) {
         // Count the total length of all of the fixed columns/rows -> totalFixed
         // Count the number of columns/rows which are fixed -> countFixed
-        if (grid[i].isFixed()) {
-            gridLayout[i] = std::max(grid[i].intValue(), 0);
+        if (grid[i].unit == HTMLDimensionsListValue::Unit::Absolute) {
+            gridLayout[i] = std::max(static_cast<int>(grid[i].number), 0);
             totalFixed += gridLayout[i];
             countFixed++;
         }
         
         // Count the total percentage of all of the percentage columns/rows -> totalPercent
         // Count the number of columns/rows which are percentages -> countPercent
-        if (grid[i].isPercentOrCalculated()) {
-            gridLayout[i] = std::max(intValueForLength(grid[i], availableLen), 0);
+        if (grid[i].unit == HTMLDimensionsListValue::Unit::Percentage) {
+            gridLayout[i] = std::max(static_cast<int>(availableLen * grid[i].number / 100.0), 0);
```

Comment:
Will the below work? Same thought for the 3 other cases below. I suppose it's nice to be able to see explicit truncation so we can ask whether we want flooring, rounding, ceiling or something else, but also would prefer not to use a cast for a conversion that works without one.
```suggestion
            gridLayout[i] = std::max<int>(availableLen * grid[i].number / 100.0, 0);
```